hydpy/models/rconc/rconc_model.py

```diff
@@ -0,0 +1,275 @@
+        >>> round_(fluxes.outflow + sum(states.sc))
+        2.0
+    """
+
+    CONTROLPARAMETERS = (rconc_control.NmbStorages, rconc_control.RecStep)
+    DERIVEDPARAMETERS = (rconc_derived.DT, rconc_derived.KSC)
+    REQUIREDSEQUENCES = (rconc_fluxes.Inflow,)
+    UPDATEDSEQUENCES = (rconc_states.SC,)
+    RESULTSEQUENCES = (rconc_fluxes.Outflow,)
+
+    @staticmethod
+    def __call__(model: modeltools.Model) -> None:
+        con = model.parameters.control.fastaccess
+        der = model.parameters.derived.fastaccess
+        flu = model.sequences.fluxes.fastaccess
+        sta = model.sequences.states.fastaccess
+        if (con.nmbstorages == 0) or modelutils.isinf(der.ksc):
+            flu.outflow = flu.inflow
+        else:
+            flu.outflow = 0.0
+            for _ in range(con.recstep):
+                sta.sc[0] += der.dt * flu.inflow
+                for j in range(con.nmbstorages - 1):
+                    d_q = min(der.dt * der.ksc * sta.sc[j], sta.sc[j])
+                    sta.sc[j] -= d_q
+                    sta.sc[j + 1] += d_q
+                j = con.nmbstorages - 1
+                d_q = min(der.dt * der.ksc * sta.sc[j], sta.sc[j])
+                sta.sc[j] -= d_q
+                flu.outflow += d_q
+
+
+class Set_Inflow_V1(modeltools.Method):
+    """Set the input for the calculation of the runoff concentration."""
+
+    RESULTSEQUENCES = (rconc_fluxes.Inflow,)
+
+    @staticmethod
+    def __call__(model: modeltools.Model, v: float) -> None:
+        flu = model.sequences.fluxes.fastaccess
+        flu.inflow = v
+
+
+class Get_Outflow_V1(modeltools.Method):
+    """Get the previously calculated runoff concentration output.
+
+    Example:
+
+        >>> from hydpy.models.rconc import *
+        >>> parameterstep("1d")
+        >>> simulationstep("12h")
+        >>> retentiontime(3.0)
+        >>> fluxes.outflow = 2.0
+        >>> model.get_outflow_v1()
+        2.0
+    """
+
+    REQUIREDSEQUENCES = (rconc_fluxes.Outflow,)
+
+    @staticmethod
+    def __call__(model: modeltools.Model) -> float:
+        flu = model.sequences.fluxes.fastaccess
+
+        return flu.outflow
+
+
+class Model(modeltools.AdHocModel):
+    """The HydPy-RConc base model."""
+
+    INLET_METHODS = ()
+    RECEIVER_METHODS = ()
+    RUN_METHODS = ()
+    INTERFACE_METHODS = (
+        Set_Inflow_V1,
+        Determine_Outflow_V1,
+        Determine_Outflow_V2,
+        Get_Outflow_V1,
+    )
+    ADD_METHODS = ()
+    OUTLET_METHODS = ()
+    SENDER_METHODS = ()
+    SUBMODELINTERFACES = ()
+    SUBMODELS = ()
+
+
+class Sub_RConcModel(modeltools.AdHocModel):
+    """Base class for submodels that comply with the submodel interfaces defined in
+    module |rconcinterfaces|."""
+
+    def get_waterbalance(self, initial_conditions: ConditionsSubmodel) -> float:
+        """Return the water balance after the submodel has been executed.
+        Requires initial conditions as parameter."""
```

Comment:
I would remove "Requires initial conditions as parameter.", as it provides no more information than already available in the method's signature.

Reply:
ok